apps/discovery_streams/lib/discovery_streams_web/channels/streaming_channel.ex

```diff
@@ -5,35 +5,31 @@ defmodule DiscoveryStreamsWeb.StreamingChannel do
     and then begins sending new data as it arrives.
   """
   use DiscoveryStreamsWeb, :channel
-  alias DiscoveryStreams.TopicSubscriber
-  alias DiscoveryStreams.TopicHelper
 
   @update_event "update"
   @filter_event "filter"
 
   intercept([@update_event])
 
   def join(channel, params, socket) do
-    topic = determine_topic(channel)
+    system_name = determine_system_name(channel)
 
-    case topic in TopicSubscriber.list_subscribed_topics() do
-      false ->
-        {:error, %{reason: "Channel #{channel} does not exist"}}
-
-      true ->
+    case Brook.get(:discovery_streams, :streaming_datasets_by_system_name, system_name) do
+      {:ok, _} ->
         send(self(), :after_join)
         {:ok, assign(socket, :filter, create_filter_rules(params))}
+
+      _ ->
+        {:error, %{reason: "Channel #{channel} does not exist"}}
     end
   end
 
-  def handle_info(:after_join, %{assigns: %{filter: filter}} = socket) do
-    push_cache_to_socket(socket, fn msg -> message_matches?(msg, filter) end)
+  def handle_info(:after_join, %{assigns: %{filter: _filter}} = socket) do
```

Comment:
Do we need to handle this anymore?

Reply:
I believe we do or we get a weird warning.  I made the signature less complicated though

Reply:
was able to nix this actually